var/spack/repos/builtin/packages/bowtie/package.py

```diff
@@ -43,7 +44,7 @@ class Bowtie(MakefilePackage):
     # See: https://github.com/BenLangmead/bowtie/issues/87, a
     # different fix is in the FreeBSD ports/package tree
     # https://svnweb.freebsd.org/ports?view=revision&revision=483954
-    patch('issue-87.patch', when='%gcc@8.0.0:')
+    patch('issue-87.patch', when='@:1.2.2 %gcc@8.0.0:')
```

Comment:
Why do we have both a conflict and a patch? Does the patch apply to all the older versions?

Reply:
Yes it confues me. I expect this patch to resolve compiling error for bowtie prior 1.2.3. However, the conflict statement prevents users from building bowtie against gcc 8. I'll test bowtie buiding later.

Reply:
I've done the following tests.

1. Build with issue-87.patch

bowtie@1.2 %gcc@8.3.0 [ok]
bowtie@1.2.0 %gcc@8.3.0 [ok]
bowtie@1.2.1 %gcc@8.3.0 [ok]
bowtie@1.2.1.1 %gcc@8.3.0 [ok]
bowtie@1.2.2 %gcc@8.3.0 [FAILED] due to compiling errors
bowtie@1.2.2_p1 %gcc@8.3.0 [FAILED] due to bulding errors
bowtie@1.2.3 %gcc@8.3.0 [FAILED] due to patching errors

2. Build WITHOUT issue-87.patch

bowtie@1.2 %gcc@8.3.0 [FAILED] due to compiling errors
bowtie@1.2.0 %gcc@8.3.0 [FAILED] due to compiling errors
bowtie@1.2.1 %gcc@8.3.0 [FAILED] due to compiling errors
bowtie@1.2.1.1 %gcc@8.3.0 [FAILED] due to compiling errors
bowtie@1.2.2 %gcc@8.3.0 [FAILED] due to compiling errors
bowtie@1.2.2_p1 %gcc@8.3.0 [FAILED] due to bulding errors
bowtie@1.2.3 %gcc@8.3.0 [ok]

So I recoomend to tighten GCC 8 conflicts. 1.2.2 and 1.2.2_p1 fail to build against GCC 8 even after patching.

```
conflicts('%gcc@8:', when='@1.2.2')
conflicts('%gcc@8:', when='@1.2.2_p1')
```

Since the content of issue-87.patch has been merged into 1.2.3, the patch will be applied onto version prior to 1.2.3:

```
patch('issue-87.patch', when='@:1.2.2 %gcc@8.0.0:')
```